modules/authentication/proc-creating-a-custom-transformer-to-provision-users-from-rhsso-to-the-software-catalog.adoc

```diff
@@ -0,0 +1,87 @@
+[id="provisioning-users-from-rhsso-to-the-software-catalog"]
+= Creating a custom transformer to provision users from Red Hat Single-Sign On (RHSSO) to the software catalog
+
+To customize how RHSSO users/groups are mapped into {product} entities, you can create a backend module that utilizes the `keycloakTransformerExtensionPoint` to provide custom user and group transformers to the Keycloak backend.
+
+.Prerequisites
+* You xref:provisioning-users-from-rhsso-to-the-software-catalog[enabled provisioning users from Red Hat Single-Sign On (RHSSO) to the software catalog].
+
+.Procedure
+. Create a new backend module with the `yarn new` command.
+
+. Add your custom user and group transformers to the `keycloakTransformerExtensionPoint`.
+
++
+Below is an example of how the backend module can be defined:
++
+.`plugins/__<module-name>__/src/module.ts`
+[source,javascript]
+----
+import {
+  GroupTransformer,
+  keycloakTransformerExtensionPoint,
+  UserTransformer,
+} from '@janus-idp/backstage-plugin-keycloak-backend';
+
+const customGroupTransformer: GroupTransformer = async (
+  entity,
+  realm,
+  groups,
+) => {
+  /* apply transformations */
+  return entity;
+};
+const customUserTransformer: UserTransformer = async (
+  entity,
+  user,
+  realm,
+  groups,
+) => {
+  /* apply transformations */
+  return entity;
+};
+
+export const keycloakBackendModuleTransformer = createBackendModule({
+  pluginId: 'catalog',
+  moduleId: 'keycloak-transformer',
+  register(reg) {
+    reg.registerInit({
+      deps: {
+        keycloak: keycloakTransformerExtensionPoint,
+      },
+      async init({ keycloak }) {
+        keycloak.setUserTransformer(customUserTransformer);
+        keycloak.setGroupTransformer(customGroupTransformer);
+        /* highlight-add-end */
+      },
+    });
+  },
+});
+----
++
+[IMPORTANT]
+====
+The `pluginId` for the module **MUST** be set to `catalog` to match the `pluginId` of the `keycloak-backend` or else the module will fail to initialize.
```

Comment:
```suggestion
The module's `pluginId` must be set to `catalog` to match the `pluginId` of the `keycloak-backend`; otherwise, the module fails to initialize.
```